core/liquidity/v2/sla.go

```diff
@@ -1,14 +1,18 @@
 package liquidity
 
 import (
+	"context"
 	"encoding/binary"
 	"math/rand"
 	"time"
 
+	"code.vegaprotocol.io/vega/core/types"
 	"code.vegaprotocol.io/vega/libs/crypto"
 	"code.vegaprotocol.io/vega/libs/num"
 )
 
+const maximumNumberOfHistoricalPenalties = 100000
```

Comment:
what's the point in keeping `maximumNumberOfHistoricalPenalties` where the maximum number of epochs ever used is performanceHysteresisEpochs, in which case you can just keep a cyclic buffer with len performanceHysteresisEpochs as you just always calculate the mean of it.

Reply:
The point is that the performanceHysteresisEpochs is a parameter that can challenge any time to larger number. And this is the max limit.

Reply:
sure, it can change, then you increase and decrease the size of the slice, so what?

Reply:
Wait. What if we keep 5 items in slice but the period increases for 30 then you miss the previous data.

Reply:
then you will be using 30 periods once you have 30 periods. I think that's sensible. It makes less sense to me to keep 100000 entries just for the odd chance that maybe at some point the community will change it from 5 to 100000.

Reply:
Well. As long as research is aligned with it. This is implemented based on their input.

Reply:
the spec doesn't say what is the valid range of this *market* parameter. What value do we expect it to have? @davidsiska-vega. I don't know. In my mind it makes sense to keep a circular buffer of size `performanceHysteresisEpochs` rather than a buffer of size up to 100k just for the case someone ever updates the market to increase the size. Which is another point - have these params been added to the spot market proposal?